### PR TITLE
Remove Conversion.Core registrations

### PR DESCRIPTION
**Pull request by JanKrivanek**
Contributes to #8826

### Context
Microsoft.Build.Conversion.Core was removed in VS codebase by https://devdiv.visualstudio.com/DevDiv/_git/VS/pullrequest/583701 However Microsoft.Build component still keeps dragging the binary to the VS install. Let's make sure Microsoft.Build.Conversion.Core.dll is not part of the VS install before we remove the code alltogether

### Changes Made
Removing the registrations of Microsoft.Build.Conversion.Core, that contributes to VS install

### CODE REVIEW

**Comment by JanKrivanek** [comment]
Experimental VS insertion PR: https://devdiv.visualstudio.com/DevDiv/_git/VS/pullrequest/585404

File: src/Package/MSBuild.VSSetup/files.swr [hunk 1]
```diff
@@ -26,9 +26,7 @@ folder InstallDir:\MSBuild\Current
   file source=$(ThirdPartyNotice)
 
 folder InstallDir:\MSBuild\Current\Bin
-  file source=$(MSBuildConversionBinPath)Microsoft.Build.Conversion.Core.dll vs.file.ngenApplications="[installDir]\Common7\IDE\vsn.exe" vs.file.ngenApplications="[installDir]\MSBuild\Current\Bin\MSBuild.exe" vs.file.ngenArchitecture=all
   file source=$(X86BinPath)Microsoft.Build.dll vs.file.ngenApplications="[installDir]\Common7\IDE\vsn.exe" vs.file.ngenApplications="[installDir]\MSBuild\Current\Bin\MSBuild.exe" vs.file.ngenArchitecture=all vs.file.ngenPriority=1
-  file source=$(MSBuildConversionBinPath)Microsoft.Build.Engine.dll vs.file.ngenApplications="[installDir]\Common7\IDE\vsn.exe" vs.file.ngenApplications="[installDir]\MSBuild\Current\Bin\MSBuild.exe" vs.file.ngenArchitecture=all
```
[the remark this answers is not in the document]
Comment (JanKrivanek) [reply src/Package/MSBuild.VSSetup/files.swr:31]: That will follow.

Engine cannot be removed from VS due to Conversions still being deployed (due to this) and depending in ngen on Engine. We need to first completely remove Coversions, then we can remove Engine

File: src/Package/MSBuild.VSSetup/MSBuild.clientenabledpkg [hunk 2]
```diff
@@ -3,5 +3,4 @@ Microsoft.Build.pkgdef
 Microsoft.Build.Tasks.Core.pkgdef
 System.Resources.Extensions.pkgdef
 Microsoft.Build.Utilities.Core.pkgdef
-Microsoft.Build.Conversion.Core.pkgdef
```
[the remark this answers is not in the document]
Comment (JanKrivanek) [reply src/Package/MSBuild.VSSetup/MSBuild.clientenabledpkg:6]: done

File: src/Package/MSBuild.VSSetup/files.swr (hunk 1 above)
[the remark this answers is not in the document]
Comment (rainersigwald) [reply src/Package/MSBuild.VSSetup/files.swr:31]: Ok then I'm confused at this part of the change that's stopping shipping Engine.

[the remark this answers is not in the document]
Comment (JanKrivanek) [reply src/Package/MSBuild.VSSetup/files.swr:31]: Oh - now I got your original question :-)
This PR could theoretically remove all of engine as well, I just wanted to limit blast radius.

In this specific case I removed the Engine, that's shipped together with Conversions.
This is not preventing other parts of VS to pull and use Engine by themselves

[the remark this answers is not in the document]
Comment (JanKrivanek) [reply src/Package/MSBuild.VSSetup/files.swr:31]: It's a fair question that I do not have good answer to. But given that the goal is anyway to get rid of both and exp insertion combining this PR with Engine removal in VS is going through without issues - I'll defer finding answer now (and I'll hope I'll not need the answer).

If anything goes sideways - I'll revert and make better due diligence in deciding the separation (or joining) of removal steps

**Pull request merged**